src/Contracts/ServiceInterface.php

```diff
@@ -11,66 +11,62 @@
     const PAGE_CHECKOUT = 'checkout';
 
     /**
-     * Code that must be added to the frontend to track user before actual purchase, some service can return null if
-     * they do not require frontend tracking
+     * Code that must be added to the frontend to track user activity before and
+     * during the purchase. Some services do not require this and will return an
+     * empty string.
      *
      * @param string $pageType One of self::PAGE_* which page is the customer on
-     *
-     * @return string|null
+     * @return string
      */
-    public function trackingCode(string $pageType): ?string;
+    public function trackingCode(string $pageType): string;
```

Comment:
I removed the nullable response type because it doesn't hurt to just always call `trackingCode($type)` without having to worry about null checking first.